pkg/plugins/plugins.go

```diff
@@ -188,6 +188,15 @@ func (pm *PluginManager) scan(pluginDir string) error {
 	return nil
 }
 
+// GetDatasource returns a datasource based on passed pluginId if it exists
+//
+// This function fetches the Datasource from the global variable in the plugins package.
```

Comment:
I don't see any reason why datasource should be capitalized here:
```suggestion
// This function fetches the datasource from the global variable DataSources in this package.
```